### DISCUSSION TITLE
StableDiffusionXLControlNetInpaintPipeline images look weird

### DISCUSSION THREAD

**Discussion by tiimgreen** [huggingface/diffusers#7504] [Q&A] [answered]
I'm using the Real Vis inpainting model (https://civitai.com/models/139562?modelVersionId=297320) which has a baked VAE. But my images are coming out like this (subject is same, BG is inpainted):

![image](https://github.com/huggingface/diffusers/assets/4130574/12b47265-a5ce-4eaa-a063-c364767d69bb)

See below for my code. Is this a VAE issue? I tried specifying `vae=None` (since the VAE is baked) but that doesn't work. Any ideas here?

```python
controlnet = ControlNetModel.from_pretrained("diffusers/controlnet-canny-sdxl-1.0")

pipe = StableDiffusionXLControlNetInpaintPipeline.from_single_file(
    "./realvisxlV40_v30InpaintBakedvae.safetensors",
    torch_dtype=torch.float16,
    controlnet=controlnet,
    use_safetensors=True,
    add_watermarker=False,
    low_cpu_mem_usage=True,
)
pipe = pipe.to("cuda")
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config, algorithm_type="sde-dpmsolver++", use_karras_sigmas=True)
```

**Comment by sayakpaul** [comment]
1. Have you tried using any other schedulers?
2. Not sure if the underlying base pipeline checkpoint would yield good results with the ControlNet checkpoint because it wasn't trained on it.

**Reply by asomoza** [reply]
I'm working with the inpainting pipeline so I tested it with the exact same model and scheduler but with my images and masks, there's nothing wrong with it:

|on a window with a lake as a background|on a table, tropical island, beach|
|---|---|
|![20240328204045_2483213170](https://github.com/huggingface/diffusers/assets/5442875/b8d212bd-9769-4c30-9a90-97fbcfe5ae17)|![20240328204230_499687826](https://github.com/huggingface/diffusers/assets/5442875/eb31eca2-5cb1-456f-989b-67b915990c52)|
|forest with leaves in the background, autumn|on the street, steampunk city at night|
|![20240328204451_944776422](https://github.com/huggingface/diffusers/assets/5442875/25ed758e-2572-4c43-b161-35e0ca3dc667)|![20240328204819_3689302423](https://github.com/huggingface/diffusers/assets/5442875/d854f24a-8ba6-4f53-a888-7e39b847dd0b)|

They're not something I would use but they are okay. So the problem is with what we don't see in your code, the image, the masks, the controlnet params or the strength.

I just noticed that I should’ve probably replied in another post.

**Reply by tiimgreen** [reply]
Ok interesting, must be something weird with my setup then. Mind sharing your code so I can cross check?

**Reply by asomoza** [reply]
yeah sure, I used everything local, so you'll need to change the urls:

```python
import torch

from diffusers import ControlNetModel, DPMSolverMultistepScheduler, StableDiffusionXLControlNetInpaintPipeline
from diffusers.utils import load_image

base = load_image("echo.png")
mask = load_image("echo_mask.png")
control_image = load_image("canny.png")

controlnet = ControlNetModel.from_pretrained(
    "controlnet-canny-sdxl-1.0-small", torch_dtype=torch.float16, variant="fp16"
)

pipe = StableDiffusionXLControlNetInpaintPipeline.from_single_file(
    "realvisxlV40_v30InpaintBakedvae.safetensors",
    torch_dtype=torch.float16,
    variant="fp16",
    controlnet=controlnet,
).to("cuda")

pipe.scheduler = DPMSolverMultistepScheduler.from_config(
    pipe.scheduler.config, algorithm_type="sde-dpmsolver++", use_karras_sigmas=True
)

mask_blurred = pipe.mask_processor.blur(mask, blur_factor=1)

prompt = "on a window with a lake as a background"

image = pipe(
    prompt,
    image=base,
    mask_image=mask_blurred,
    guidance_scale=6.5,
    strength=1.0,
    num_inference_steps=25,
    control_image=control_image,
    controlnet_conditioning_scale=0.8,
    control_guidance_end=0.7,
).images[0]
```

I did a better mask because I didn't like the results ^^. Now it looks a lot better:

![20240329000930_3733138106](https://github.com/huggingface/diffusers/assets/5442875/5ab63210-aff3-4e3d-90c0-af5529484ab8)

Do you want the images too?

**Reply by tiimgreen** [reply]
Ok I tried exactly those settings and got the same problem. So either my images are weird somehow or its the machine I'm running it on.

Could you send your images please?

Also how are you running this? I'm on RunPod using the `runpod/pytorch:2.0.1-py3.10-cuda11.8.0-devel-ubuntu22.04` image

**Reply by asomoza** [reply]
I did the last one on windows, the firsts ones were on linux, I switch a lot. In windows I use pytorch 2.1.2 and in linux pytorch 2.2.2 both with cuda 12.1

Here are the images, you can click on them to get the full size image.

|source|mask|canny|
|---|---|---|
|![pexels-anete-lusina-4790268](https://github.com/huggingface/diffusers/assets/5442875/e0b20976-c3d9-444c-be8d-91ba81cd75e6)|![20240328200118](https://github.com/huggingface/diffusers/assets/5442875/6fef0c8e-9f87-46b5-9cef-618a6aa3c05e)|![echo_mask](https://github.com/huggingface/diffusers/assets/5442875/0d29f450-9f51-4a8a-9c59-497f50fa1f64)|

**Reply by tiimgreen** [reply]
Ok with your images it works. I see that yours are 1024 and mine are 512, could be that? Or maybe some weird encoding stuff?

I'll do some digging though, this was helpful. Thanks!

**Reply by asomoza** [reply, marked as answer]
yeah, that's it, SDXL works with 1024, don't use 512 with SDXL unless you use the turbo model.

**Reply by tiimgreen** [reply]
Ah ok, I was migrating from a 1.5 model, didn't think to check that. Thanks for your help
